frontend/src/components/common/UploadedFilesList.js

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import moment from "moment";
+import PropTypes from "prop-types";
+import { Col, Row, Popconfirm, Icon } from "antd";
+import { bindActionCreators } from "redux";
+import { connect } from "react-redux";
+
+import CustomPropTypes from "@/customPropTypes";
+import {
+  removeMineDocumentFromExpectedDocument,
+  fetchMineRecordById,
+} from "@/actionCreators/mineActionCreator";
+import { getMines } from "@/selectors/mineSelectors";
+import { getExpectedDocumentStatusOptions } from "@/selectors/staticContentSelectors";
+
+const propTypes = {
+  selectedDocId: PropTypes.string.isRequired,
+  mineId: PropTypes.string.isRequired,
+  removeMineDocumentFromExpectedDocument: PropTypes.func.isRequired,
+  fetchMineRecordById: PropTypes.func.isRequired,
+  mines: PropTypes.objectOf(CustomPropTypes.mine).isRequired,
+  expectedDocumentStatusOptions: PropTypes.arrayOf(CustomPropTypes.dropdownListItem).isRequired,
+};
+
+export class UploadedFilesList extends React.Component {
+  unlinkFile = (event, mineDocumentGuid, expDocumentGuid) => {
+    event.preventDefault();
+    this.props
+      .removeMineDocumentFromExpectedDocument(mineDocumentGuid, expDocumentGuid)
+      .then(this.props.fetchMineRecordById(this.props.mineId));
+  };
+
+  canUnlink = (selectedDoc) => {
+    const docStatus = this.props.expectedDocumentStatusOptions.find(
+      (status) => status.value === selectedDoc.exp_document_status_guid
+    );
+
+    if (moment(selectedDoc.due_date, "YYYY-MM-DD") > moment()) {
+      if (docStatus.label === "Not Received" || docStatus.label === "Received / Pending Review") {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  render() {
+    const selectedDoc = this.props.mines[this.props.mineId].mine_expected_documents.find(
+      (expDoc) => expDoc.exp_document_guid === this.props.selectedDocId
+    );
+    const unlink = this.canUnlink(selectedDoc);
+    return (
+      <div>
+        {selectedDoc.related_documents.map((file, key) => (
```

Comment:
I don't have a super strong opinion on this, and I can see why you called it `key`, but I think it may be best to use the documented name here (`i`, `index`, or `idx`) since that's what this param actually is in the context of a `map`